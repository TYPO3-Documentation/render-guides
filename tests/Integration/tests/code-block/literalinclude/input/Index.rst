==============
Document Title
==============

This is a paragraph above.

..  literalinclude:: _code/_Example.php
    :caption: Some PHP

..  literalinclude:: _code/_setup.typoscript
    :caption: Some TypoScript

..  literalinclude:: _code/_test.txt
    :caption: Some Plaintext

..  literalinclude:: _code/_Template.html
    :caption: Some Fluid

..  literalinclude:: _code/_composer.json
    :caption: Some JSON

..  literalinclude:: _code/_Services.yaml
    :caption: Some YAML

..  literalinclude:: _code/_some.yml
    :caption: Some YAML

..  literalinclude:: _code/_some.diff
    :caption: Some diff

..  literalinclude:: _code/_Example.php
    :caption: Language property takes precedence
    :language: plaintext

This is a paragraph below.
