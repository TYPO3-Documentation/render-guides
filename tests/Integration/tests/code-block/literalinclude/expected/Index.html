<!-- content start -->
                <section class="section" id="document-title">
            <h1>Document Title<a class="headerlink" href="#document-title" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">Â¶</a></h1>

    <p>This is a paragraph above.</p>

                <div class="code-block-caption">
        <span class="caption-text">Some PHP</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs php"><span class="hljs-meta">&lt;?php</span>

<span class="hljs-keyword">declare</span>(strict_types=<span class="hljs-number">1</span>);

<span class="hljs-class"><span class="hljs-keyword">class</span> <span class="hljs-title">Example</span>
</span>{
    <span class="hljs-keyword">public</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">test</span><span class="hljs-params">()</span>: <span class="hljs-title">string</span>
    </span>{
        <span class="hljs-keyword">return</span> <span class="hljs-string">'this is a test'</span>;
    }
}
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_Example.php" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some TypoScript</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs typoscript">page = <span class="hljs-keyword">PAGE</span>
page.<span class="hljs-title">10</span> = <span class="hljs-keyword">TEXT</span>
page.<span class="hljs-title">10</span>.value = Hello World!
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_setup.typoscript" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some Plaintext</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs plaintext">Hello World!
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_test.txt" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some Fluid</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs xml"><span class="hljs-tag">&lt;<span class="hljs-name">f:debug</span>&gt;</span>{_all}<span class="hljs-tag">&lt;/<span class="hljs-name">f:debug</span>&gt;</span>
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_Template.html" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some JSON</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs json">{
    <span class="hljs-attr">"name"</span>: <span class="hljs-string">"hello/world"</span>
}
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_composer.json" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some YAML</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs yaml"><span class="hljs-attr">services:</span>
  <span class="hljs-attr">_defaults:</span>
    <span class="hljs-attr">autowire:</span> <span class="hljs-literal">true</span>
    <span class="hljs-attr">autoconfigure:</span> <span class="hljs-literal">true</span>
    <span class="hljs-attr">public:</span> <span class="hljs-literal">false</span>

</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_Services.yaml" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some YAML</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs yaml"><span class="hljs-attr">hello:</span> <span class="hljs-string">world</span>
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_some.yml" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Some diff</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs diff">Hello World
<span class="hljs-deletion">- Hello Typo 3</span>
<span class="hljs-addition">+ Hello TYPO3</span>
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_some.diff" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
                <div class="code-block-caption">
        <span class="caption-text">Language property takes precedence</span>
    </div><div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs plaintext">&lt;?php

declare(strict_types=1);

class Example
{
    public function test(): string
    {
        return 'this is a test';
    }
}
</code></pre>
    <div>            <a href="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/edit/documentation-draft/Documentation/_code/_Example.php" class="code-block-edit" title="Edit on GitHub">
                <i class="icon fa-brands fa-github"></i>
            </a>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>

    <p>This is a paragraph below.</p>

    </section>
        <!-- content end -->
